src/crimson/os/seastore/omap_manager/btree/string_kv_node_layout.h

```diff
@@ -581,13 +581,20 @@ class StringKVInnerNodeLayout {
   }
 
   const_iterator string_upper_bound(std::string_view str) const {
-    auto ret = iter_begin();
-    for (; ret != iter_end(); ++ret) {
-      std::string s = ret->get_key();
-      if (s > str)
-        break;
+    uint16_t start = 0, end = get_size();
+    while (start != end) {
+      unsigned mid = (start + end) / 2;
+      const_iterator iter(this, mid);
+      std::string s = iter->get_key();
+      if (s < str) {
+        start = ++mid;
+      } else if ( s > str) {
+        end = mid;
+      } else {
+        return ++iter;
+      }
```

Comment:
instead of implementing binary search manually, how about using `std::binary_search()` with a customized comparator? IMHO, more readable this way.

Reply:
since std::binary_search only return bool, I just use std::lower_bound and std::upper_bound to implement it by customized comparator. Please review again.